gnome-image-installer/util/gis-unattended-config.c

```diff
@@ -0,0 +1,175 @@
+                        gis_unattended_config,
+                        G_TYPE_OBJECT,
+                        0,
+                        G_IMPLEMENT_INTERFACE (G_TYPE_INITABLE,
+                                               gis_unattended_config_initable_iface_init));
+
+typedef enum {
+  PROP_FILE_PATH = 1,
+  N_PROPERTIES
+} GisUnattendedConfigPropertyId;
+
+static GParamSpec *props[N_PROPERTIES] = { 0 };
+
+static void
+gis_unattended_config_init (GisUnattendedConfig *self)
+{
+  self->key_file = g_key_file_new ();
+}
+
+static void
+gis_unattended_config_set_property (GObject      *object,
+                                    guint         property_id,
+                                    const GValue *value,
+                                    GParamSpec   *pspec)
+{
+  GisUnattendedConfig *self = GIS_UNATTENDED_CONFIG (object);
+
+  switch ((GisUnattendedConfigPropertyId) property_id)
+    {
+    case PROP_FILE_PATH:
+      g_free (self->file_path);
+      self->file_path = g_value_dup_string (value);
+      break;
+
+    case N_PROPERTIES:
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+      break;
+    }
+}
+
+static void
+gis_unattended_config_dispose (GObject *object)
+{
+  GisUnattendedConfig *self = GIS_UNATTENDED_CONFIG (object);
+
+  g_clear_pointer (&self->key_file, g_key_file_unref);
+
+  G_OBJECT_CLASS (gis_unattended_config_parent_class)->dispose (object);
+}
+
+static void
+gis_unattended_config_finalize (GObject *object)
+{
+  GisUnattendedConfig *self = GIS_UNATTENDED_CONFIG (object);
+
+  g_clear_pointer (&self->file_path, g_free);
+
+  G_OBJECT_CLASS (gis_unattended_config_parent_class)->finalize (object);
+}
+
+static void
+gis_unattended_config_class_init (GisUnattendedConfigClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->set_property = gis_unattended_config_set_property;
+  // object_class->get_property = gis_unattended_config_get_property;
+  object_class->dispose = gis_unattended_config_dispose;
+  object_class->finalize = gis_unattended_config_finalize;
+
+  props[PROP_FILE_PATH] = g_param_spec_string (
+      "file-path",
+      "file path",
+      "Path to unattended.ini file",
+      NULL,
+      G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+
+  g_object_class_install_properties (object_class, N_PROPERTIES, props);
+}
+
+static gboolean
+gis_unattended_config_initable_init (GInitable    *initable,
+                                     GCancellable *cancellable,
+                                     GError      **error)
+{
+  GisUnattendedConfig *self = GIS_UNATTENDED_CONFIG (initable);
+  g_autoptr(GError) local_error = NULL;
+
+  if (self->file_path == NULL)
+    {
+      g_set_error (error,
+                   G_FILE_ERROR,
+                   G_FILE_ERROR_INVAL,
+                   "programmer error: GisUnattendedConfig:file-path must "
+                   "not be NULL");
```

Comment:
If it’s a programmer error, use `g_return_val_if_fail()` rather than `GError`.

Reply:
Ok